File: submissions/0014-longest-common-prefix/solution.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        int num = strs.length;
        if (num == 0) {
            return "";
        }
        if (num == 1) {
            return strs[0];
        }
        char[][] charsArr = new char[num][];

        for (int i = 0; i < strs.length; i++) {
            charsArr[i] = strs[i].toCharArray();
        }
        int j = 0;
        boolean looping = true;
        while (looping) {
            if (charsArr[0].length < j + 1) {
                break;
            } else {
                char match = charsArr[0][j];
                for (int i = 1; i < num; i++) {
                    if (charsArr[i].length < j + 1) {
                        looping = false;
                        break;
                    } else {
                        if (charsArr[i][j] != match) {
                            looping = false;
                            break;
                        }
                    }
                }
                if (looping) {
                    j++;
                }
            }
        }
        if (j == 0) {
            return "";
        } else {
            return strs[0].substring(0, j);
        }
    }
}

File: submissions/0014-longest-common-prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        minimumLength = min([len(s) for s in strs])
        if minimumLength == 0:
            return ""
        
        if len(strs) == 1:
            return strs[0]
        
        for i in range(minimumLength):
            prefixes = [s[i] for s in strs]
            # if prefixes are not all identitcal
            if len(set(prefixes)) != 1:
                return strs[0][:i]

        return strs[0][:minimumLength]
        
